specification/app/resource-manager/Microsoft.App/preview/2023-11-02-preview/JavaComponents.json

```diff
@@ -0,0 +1,459 @@
+            "headers": {
+              "Location": {
+                "type": "string"
+              }
+            }
+          },
+          "204": {
+            "description": "Java Component does not exist."
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-examples": {
+          "Delete Java Component": {
+            "$ref": "./examples/JavaComponents_Delete.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "JavaComponentProperty": {
+      "description": "Configuration properties for a Java Component",
+      "type": "object",
+      "properties": {
+        "propertyName": {
+          "description": "The name of the property",
+          "type": "string"
+        },
+        "value": {
+          "description": "The value of the property",
+          "type": "string"
+        }
+      }
+    },
+    "JavaComponentServiceBind": {
+      "description": "Configuration to bind a Java Component to another Java Component",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Name of the service bind",
+          "type": "string"
+        },
+        "serviceId": {
+          "description": "Resource id of the target service",
+          "type": "string"
+        }
+      }
+    },
+    "JavaComponent": {
+      "description": "Java Component.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Java Component resource specific properties",
+          "type": "object",
+          "properties": {
+            "componentType": {
+              "description": "Type of the Java Component.",
+              "type": "string"
+            },
+            "provisioningState": {
+              "description": "Provisioning state of the Java Component.",
+              "enum": [
+                "Succeeded",
+                "Failed",
+                "Canceled",
+                "Deleting",
+                "InProgress"
+              ],
+              "type": "string",
+              "readOnly": true,
+              "x-ms-enum": {
+                "name": "JavaComponentProvisioningState",
+                "modelAsString": true
+              }
+            },
+            "configurations": {
+              "description": "List of Java Components configuration properties",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/JavaComponentProperty"
```

Comment:
>JavaComponentProperty [](http://example.com/codeflow?start=39&length=21)

Should this be named as "JavaComponentConfigurationProperty"?

Reply:
Make sense, the name is changed.